spring-data-eclipse-store/src/main/java/software/xdev/spring/data/eclipse/store/repository/lazy/SpringDataEclipseStoreLazyBinaryHandler.java

```diff
@@ -0,0 +1,195 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package software.xdev.spring.data.eclipse.store.repository.lazy;
+
+import java.lang.reflect.Constructor;
+import java.util.Objects;
+
+import org.eclipse.serializer.memory.XMemory;
+import org.eclipse.serializer.persistence.binary.types.AbstractBinaryHandlerCustom;
+import org.eclipse.serializer.persistence.binary.types.Binary;
+import org.eclipse.serializer.persistence.binary.types.BinaryTypeHandler;
+import org.eclipse.serializer.persistence.types.PersistenceLoadHandler;
+import org.eclipse.serializer.persistence.types.PersistenceReferenceLoader;
+import org.eclipse.serializer.persistence.types.PersistenceStoreHandler;
+import org.eclipse.serializer.reference.Lazy;
+import org.eclipse.serializer.reference.ObjectSwizzling;
+import org.eclipse.serializer.reflect.XReflect;
+
+import software.xdev.spring.data.eclipse.store.repository.support.copier.working.WorkingCopier;
+
+
+/**
+ * This is a complicated one. First off: this handler should only be used for WorkingCopies (see
+ * {@link software.xdev.spring.data.eclipse.store.repository.support.copier.copier.EclipseSerializerRegisteringCopier})!
+ * <p>
+ *     First case:<br/>
+ *     The user creates a {@link SpringDataEclipseStoreLazy} and puts a object in it.
+ *     This object is stored as with a default {@link BinaryTypeHandler}. But when it gets loaded,
+ *     it <b>does not</b> load as the stored object, but it gets wrapped in a {@link Lazy#Reference(Object)}.
+ * </p>
+ * <p>
+ *     Second case:<br/>
+ *     The actual lazy object gets loaded from the actual storage. In this case the {@link ObjectSwizzling} is
+ *     important! It's the actual {@link ObjectSwizzling} from the storage (not from the
+ * {@link software.xdev.spring.data.eclipse.store.repository.support.copier.copier.EclipseSerializerRegisteringCopier}).
+ *     This means, the {@link SpringDataEclipseStoreLazy} holds the objectId of the original lazy in the original
+ *     storage.
+ *     Therefore if {@link SpringDataEclipseStoreLazy#get()} is called a new working copy of the lazy from the
+ *     storage is loaded.
+ * </p>
+ */
+public final class SpringDataEclipseStoreLazyBinaryHandler
+	extends AbstractBinaryHandlerCustom<SpringDataEclipseStoreLazy.Default<?>>
+{
+	@SuppressWarnings("rawtypes")
+	static final Constructor<SpringDataEclipseStoreLazy.Default> CONSTRUCTOR_SURROGATE_LAZY = XReflect.setAccessible(
+		XReflect.getDeclaredConstructor(
+			SpringDataEclipseStoreLazy.Default.class,
+			long.class,
+			ObjectSwizzling.class,
+			WorkingCopier.class
+		)
+	);
+	
+	public static final int OFFSET_UNWRAPPED_OBJECT = 8;
```

Comment:
`public` visibility necessary?

Reply:
Are those constants maybe available from eclipsestore?

Reply:
I think they are available, but should not be used. This is a pattern, that is common in the binary handler and should only be used locally. The value could change in the future and then should also only affect this class.